### DISCUSSION TITLE
[Feature] Zip-Import (.zip, .tar, .rar, .7z)

### DISCUSSION THREAD

**Discussion by MickLesk** [immich-app/immich#8746] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hello dear development team,

it would be a really good feature if something like a zip import via the upload was possible.

This would allow you, for example, to zip all your vacation photos once and upload them as a complete file. 

Logic:
-> Zip/Rar/7z/Tar (the common formats) is selected via upload
-> Upload dialog appears (until complete or error)
-> The zip file is unpacked by immich and 
1. either directly assigned to the correct album (if you are in one)
OR
2. if there is a folder, the album is created after the folder
OR
3. or it is simply stored normally in Photos

=> Maybe you can also solve the whole thing with a dialog in which you can either say "add to album XYZ", or "upload only", or "create folder as album" (is only shown if there is at least one folder in the zip), If there are several folders in the zip and the check mark is set, several albums are of course created. The action "add to album XYZ" sets the other action readonly, then everything is added to the selected album regardless of the folder

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by milankkk** [comment]
i second this

**Comment by PabloAM50** [comment]
Is this not implemented yet?????

**Comment by millionmice** [comment]
Even just allowing external libraries to have their archives scanned and included would be helpful.
